addons/web_editor/static/src/js/wysiwyg/widgets/image_crop_widget.js

```diff
@@ -1,174 +1,238 @@
-odoo.define('wysiwyg.widgets.CropImageDialog', function (require) {
+odoo.define('wysiwyg.widgets.ImageCropWidget', function (require) {
 'use strict';
 
-var core = require('web.core');
-var Dialog = require('wysiwyg.widgets.Dialog');
+const core = require('web.core');
+const qweb = core.qweb;
+const Widget = require('web.Widget');
 
-var _t = core._t;
+const _t = core._t;
 
-/**
- * CropImageDialog widget. Let users crop an image.
- */
-var CropImageDialog = Dialog.extend({
-    template: 'wysiwyg.widgets.crop_image',
-    xmlDependencies: Dialog.prototype.xmlDependencies.concat(
-        ['/web_editor/static/src/xml/wysiwyg.xml']
-    ),
-    events: _.extend({}, Dialog.prototype.events, {
-        'click .o_crop_options [data-event]': '_onCropOptionClick',
-    }),
+// Fields returned by cropper lib 'getData' method
+const cropperDataFields = ['x', 'y', 'width', 'height', 'rotate', 'scaleX', 'scaleY'];
+const ImageCropWidget = Widget.extend({
+    xmlDependencies: ['/web_editor/static/src/xml/wysiwyg.xml'],
+    // Crop attributes that are saved with the DOM. Should only be removed when the image is changed.
+    persistentAttributes: [
+        ...cropperDataFields,
+        'aspectRatio',
+    ],
+    // Attributes that are used to keep data from one crop to the next in the same session
+    // If present, should be used by the cropper instead of querying db
+    sessionAttributes: [
+        'attachmentId',
+        'originalSrc',
+        'originalId',
+        'originalName',
+        'mimetype',
+    ],
+    // Attributes that are used by saveCroppedImages to create or update attachments
+    saveAttributes: [
+        'resModel',
+        'resId',
+        'attachmentId',
+        'originalId',
+        'originalName',
+        'mimetype',
+    ],
 
     /**
      * @constructor
      */
-    init: function (parent, options, media) {
-        var self = this;
+    init(parent, options, media) {
+        this._super(...arguments);
         this.media = media;
-        this.$media = $(this.media);
-        var src = this.$media.attr('src').split('?')[0];
-        this.aspectRatioList = [
-            [_t("Free"), '0/0', 0],
-            ["16:9", '16/9', 16 / 9],
```

Comment:
Feels like the logic of the second value is gone in your PR. You are in your web comparing a float parsed from a data attribute with a computed value. I am pretty sure that `parseFloat('1.33333333333333') === (4/3)` is false for any acceptable number of threes to save in the DOM ;)